### PR TITLE
Clean no-op code in `linkWithEmailPassword` method

### PR DESCRIPTION
**Pull request by renkelvin**

### CODE REVIEW

File: FirebaseAuth/Sources/User/FIRUser.m [hunk 1]
```diff
@@ -1129,8 +1129,6 @@ - (void)linkWithEmailPassword:(FIREmailPasswordAuthCredential *)credential
                         callInMainThreadWithAuthDataResultAndError(completion, nil, keychainError);
                         return;
                       }
-                      [self signOutIfTokenIsInvalidWithError:error];
```
Comment (paulb777) [inline FirebaseAuth/Sources/User/FIRUser.m:1132]: Maybe this signOut should be in the if block above?

File: FirebaseAuth/Sources/User/FIRUser.m [hunk 2]
```diff
@@ -1129,8 +1129,6 @@ - (void)linkWithEmailPassword:(FIREmailPasswordAuthCredential *)credential
...
-                      callInMainThreadWithAuthDataResultAndError(completion, authResult, nil);
```
Comment (paulb777) [inline FirebaseAuth/Sources/User/FIRUser.m:1133]: Isn't this the success callback to the caller?

File: FirebaseAuth/Sources/User/FIRUser.m (hunk 1 above)
Comment (renkelvin) [reply FirebaseAuth/Sources/User/FIRUser.m:1132]: I'm still working on this. Wait a bit. 😂

Comment (renkelvin) [reply FirebaseAuth/Sources/User/FIRUser.m:1132]: Confirmed that it's not necessary. `updateKeychain` should not raise errors of invalid token.

File: FirebaseAuth/Sources/User/FIRUser.m (hunk 2 above)
Comment (renkelvin) [reply FirebaseAuth/Sources/User/FIRUser.m:1133]: Yes, correct. Tested e2e and it should work fine now.

**Review by paulb777 — APPROVED** [review]

**Pull request merged**